### PR TITLE
Delete code not-related to dispute tools, use Discourse as SSO provider

### PR DESCRIPTION
**Pull request by marcondag**

### CODE REVIEW

File: controllers/DisputeToolsController.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
-/* globals RestfulController, DisputeTool, neonode, Class, CONFIG, Dispute */
+/* globals RestfulController, DisputeTool, Class */
 const marked = require('marked');
+const authenticate = require('../services/authentication');
+const authorization = require('../services/authorization');
```
Comment (shoshber) [inline controllers/DisputeToolsController.js]: can we make the difference between "authentication" and "authorization" clearer (changing the filenames/classnames)?

File: controllers/DisputeToolsController.js [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+          try {
+            const disputeTool = await DisputeTool.findById(req.params.id);
+            if (disputeTool) {
               res.locals.disputeTool = disputeTool;
-
               next();
-            })
-            .catch(next);
+            } else {
+              throw new Error('Not found!');
```
Comment (shoshber) [inline controllers/DisputeToolsController.js]: A more descriptive error message would be nice here (unless the error handling infrastructure provides more info somewhere else--I haven't dug in to that much)

File: package.json [hunk 3]
```diff
@@ -92,6 +92,7 @@
     "im-metadata": "^3.0.0",
     "imagemagick-stream": "^4.1.1",
     "imagesize": "^1.0.0",
+    "jsonwebtoken": "^8.1.1",
```
Comment (shoshber) [inline package.json]: I bet there's a ton more dependencies we can get rid of

File: tests/unit/services/authentication.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const authenticate = require('../../../services/authentication');
+
+describe('authentication', () => {
+  const noop = () => {};
+
+  describe('by cookie', () => {
+    xit('should set the user using the cookie', () => {});
```
Comment (shoshber) [inline tests/unit/services/authentication.js]: `it.skip` is preferred & more readable (https://github.com/mochajs/mocha/issues/596#issuecomment-13806105)

File: controllers/DisputeToolsController.js (hunk 1 above)
Comment (marcondag) [reply controllers/DisputeToolsController.js]: Hmmm... these are pretty traditional names, authentication for getting the user of the request, authorization to see if the user can do what they're asking. If you have some better names I'm happy to change it, though.

File: controllers/DisputeToolsController.js (hunk 2 above)
Comment (marcondag) [reply controllers/DisputeToolsController.js]: Yes, the error handling is very much half-baked for this.

File: tests/unit/services/authentication.js (hunk 4 above)
Comment (marcondag) [reply tests/unit/services/authentication.js]: Oh, good to know. None of the tests will be skipped once this is actually read to merge in. Still WIP.

File: package.json (hunk 3 above)
Comment (marcondag) [reply package.json]: I'll take a closer look once I wrap up the SSO integration, you're probably right.

File: controllers/DisputeToolsController.js (hunk 1 above)
Comment (marcondag) [reply controllers/DisputeToolsController.js]: Thought about this overnight. How about:

`authenticate` -> `requireAuthentication`
`authorization` -> `authorizeWhen`

Comment (shoshber) [reply controllers/DisputeToolsController.js]: If they are traditional, we can keep them and I'll just shake my fist at the universe

**Pull request merged**